src/Uno.UI/UI/Xaml/Controls/TextBlock/JavaStringCache.android.cs

```diff
@@ -0,0 +1,189 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Text;
+using Windows.Foundation;
+
+using Uno;
+using Uno.Extensions;
+using Uno.UI;
+using Uno.Foundation.Logging;
+using Windows.UI.Xaml.Media;
+using Uno.Collections;
+using Android.Security.Keystore;
+using Java.Security;
+using Uno.Buffers;
+using Windows.System;
+
+namespace Windows.UI.Xaml.Controls
+{
+	/// <summary>
+	/// A TextBlock measure cache for non-formatted text.
+	/// </summary>
+	internal static class JavaStringCache
+	{
+		private static Logger _log = typeof(JavaStringCache).Log();
+		private static Stopwatch _watch = Stopwatch.StartNew();
+		private static HashtableEx _table = new();
+		private static TimeSpan _lastScavenge;
+		private static object _gate = new();
+
+		internal static readonly TimeSpan LowMemoryTrimInterval = TimeSpan.FromMinutes(5);
+		internal static readonly TimeSpan MediumMemoryTrimInterval = TimeSpan.FromMinutes(3);
+		internal static readonly TimeSpan HighMemoryTrimInterval = TimeSpan.FromMinutes(1);
+		internal static readonly TimeSpan OverLimitMemoryTrimInterval = TimeSpan.FromMinutes(.5);
+
+		internal static readonly TimeSpan ScavengeInterval = TimeSpan.FromMinutes(.5);
+
+		private static DefaultArrayPoolPlatformProvider _platformProvider = new DefaultArrayPoolPlatformProvider();
+
+		/// <summary>Determines if automatic memory management is enabled</summary>
+		private static readonly bool _automaticManagement;
+		/// <summary>Determines if GC trim callback has been registerd if non-zero</summary>
+		private static int _trimCallbackCreated;
+
+		private record KeyEntry(string Value, Java.Lang.String NativeValue)
+		{
+			public TimeSpan LastUse { get; set; } = _watch.Elapsed;
+		}
+
+		static JavaStringCache()
+		{
+			_automaticManagement = WinRTFeatureConfiguration.ArrayPool.EnableAutomaticMemoryManagement && _platformProvider.CanUseMemoryManager;
```

Comment:
Actually, do we guarantee that this is executed *after* the user changed `WinRTFeatureConfiguration.ArrayPool.EnableAutomaticMemoryManagement`? I think `_automaticManagement` might be better as a property and execute the getter every time the value is read so that we can detect changes of `EnableAutomaticMemoryManagement`

Reply:
There's no need to have this dynamically enabled at this point (managing the state of the change would not be useful). Most feature configuration behave this way, and this one is not different.